src/shared/src/types.rs

```diff
@@ -1,10 +1,22 @@
 use candid::{CandidType, Deserialize, Principal};
 use std::fmt::Debug;
 
+#[derive(CandidType, Deserialize)]
+pub struct SupportedCredential {
+    pub credential_type: String,
+    pub ii_origin: String,
```

Comment:
What happens if we support both `ic0.app` and `internetcomputer.org`? Not an issue with a single `ii_origin`?

Reply:
Not an issue, the credentials from both will have the same value.

I believe this is again a naming confusion.

As an example you can try to the VC flow with the dummy [relying party](https://l7rua-raaaa-aaaap-ahh6a-cai.icp0.io/) and dummy issuer (default values).

After receiving the credentials, you can decode them and then decode the alias credential.

The "iss" field is what this `ii_origin` means. It's the issuer.

![Screenshot 2024-07-16 at 13 13 48](https://github.com/user-attachments/assets/617a0dc5-4c2b-473e-94f4-10a5729d61e6)

I used this names (issuer_origin also) because it's the name used i[n the library](https://github.com/dfinity/verifiable-credentials-sdk/blob/a116a2e53a13e4a1f6f7e8429960a7d0d6142e70/rust-packages/ic-verifiable-credentials/src/lib.rs#L45) that Bartosz created to validate the credentials.

I'm ok changing the name here and then mapping it properly. I also found it confusing from the coding perspective, but it made sense from the standard perspective.

Reply:
I'm not sure I still get it but, thanks for the explanation. I think you are right we should use the naming convetion even if they do feel a bit weird at first sight.

That said, I'm still unsure what happens when a credential is issued on iss ic0.app but used on internetcomputer.org but, definitely nothin to do with this PR.

Reply:
internetcomputer.org will issue the same credential on ic0.app because both have the identity.ic0.app hardcoded.